terraso_backend/apps/core/management/commands/loadbackup.py

```diff
@@ -0,0 +1,275 @@
+            storage.bucket.meta.client.download_fileobj(bucket_name, path, temp_file)
+        return tmp_file_path
+
+    @classmethod
+    def _find_last_backup_s3(cls):
+        storage = S3BackupStorage()
+        _dirs, files = storage.listdir(".")
+        files.sort()
+        data_file, migrations_file = [
+            cls._copy_s3_file(storage, path, suffix)
+            for path, suffix in zip(files[-2:], (".json.gz", ".json"))
+        ]
+        return data_file, migrations_file
+
+    def handle(self, *args, **options):
+
+        try:
+            if user_id := options.get("save_user"):
+                user = User.objects.get(id=user_id)
+            else:
+                user = None
+            pass
+            if session_pk := options.get("save_session"):
+                session = Session.objects.get(session_key=session_pk)
+            else:
+                session = None
+
+        except Exception:
+            msg = "Error accessing user and session"
+            logger.exception(msg)
+            raise CommandError(msg)
+
+        try:
+            if options["s3"]:
+                data, migrations_file = self._find_last_backup_s3()
+            else:
+                data, migrations_file = self._find_latest_backup_dir(options["directory"])
+            with open(migrations_file, "rb") as fp:
+                migrations = json.load(fp)
+        except Exception:
+            msg = "Failure loading backup files"
+            logger.exception(msg)
+            raise CommandError(msg)
+
+        def cleanup():
+            if options["s3"]:
+                for path in data, migrations_file:
+                    os.unlink(path)
+
+        with connection.cursor() as cursor:
+            try:
+                # background task progress is saved in a DB table
+                # need to make sure this table is not deleted. otherwise things will break
+                cursor.execute(
+                    "SELECT tablename FROM pg_tables WHERE schemaname = 'public' "
+                    "AND tablename NOT IN ('core_backgroundtask', 'django_session');"
+                )
+                tables_to_drop = cursor.fetchall()
+                if session_pk:
+                    cursor.execute(
+                        "DELETE FROM django_session WHERE session_key != %s", (str(session_pk),)
+                    )
+                else:
+                    cursor.execute("DELETE FROM django_session")
+                for (table,) in tables_to_drop:
+                    cursor.execute(
+                        sql.SQL("DROP TABLE IF EXISTS {} CASCADE").format(sql.Identifier(table))
+                    )
+            except Exception:
+                msg = "Command failed resetting database"
+                logger.exception(msg)
+                connection.rollback()
+                cleanup()
+                raise CommandError(msg)
+
+            connection.commit()
+
+        try:
+            core_skip = [23, 25]
```

Comment:
These migrations need to be skipped. `23` adds the background tasks table, but this table needs to not be deleted. `25` adds taxonomy terms that are already present in the data restore. In the future, we can add other migrations that need to be skipped here.